### DISCUSSION TITLE
Where is $this->model->hydrate defined ?

### DISCUSSION THREAD

**Discussion by felahdab** [laravel/framework#48253] [Q&A] [answered]
I am probably going to look stupid or something but I can't find where that method is defined...

I am tracking what could be a bug (trying to extend an existing Model like in [this article of Caleb Porzio](https://tighten.com/insights/extending-models-in-eloquent/) in order to better separate concerns).

I have tracked down my problem down to this line of code in Illuminate\Database\Eloquent\Builder which apparently makes the application die :
[Source](https://github.com/laravel/framework/blob/4989e6de076688ade265e2f1970ab6f0c1b60fcb/src/Illuminate/Database/Eloquent/Builder.php#L736)

I would appreciate any help about that...
Regards.
Florian.

**Comment by ahmedash95** [comment]
Hello @felahdab,

It's defined in the Builder class [src/Illuminate/Database/Eloquent/Builder.php](https://github.com/laravel/framework/blob/10.x/src/Illuminate/Database/Eloquent/Builder.php#L405). 

My way to navigate through the framework source code is by using PHPStorm. and `File structure` feature under the menu Navigate -> File structure. and don't forget to check Inherited members
![image](https://github.com/laravel/framework/assets/8272048/cff17568-76e1-49ee-84b5-01feace560ef)

**Reply by henzeb** [reply]
yes. but that's not the problem. it calls $this->model->hydrate(). according the docblock that property should contain a Model instance. a Model does not have a hydrate method. That's the problem.

**Reply by ahmedash95** [reply, marked as answer]
The model has a forward call proxy that passes some calls to the QueryBuilder class. hence the hydrate method is accessable trough the model instance

https://github.com/laravel/framework/blob/4989e6de076688ade265e2f1970ab6f0c1b60fcb/src/Illuminate/Database/Eloquent/Model.php#L2333

**Reply by felahdab** [reply]
Thanks guys.
I didn't think of __call...

Regards.

**Reply by henzeb** [reply]
The construction of it so weird. The first if statement in that __call method makes you raise your eyebrows...
